apps/user-service/src/auth/reset-password.service.ts

```diff
@@ -0,0 +1,81 @@
+import { Injectable, NotFoundException, BadRequestException, Logger } from "@nestjs/common";
+import bcrypt from 'bcryptjs';
+import { JwtService } from '@nestjs/jwt';
+import { User } from '@terramatch-microservices/database/entities';
+import { EmailService } from "@terramatch-microservices/common/email/email.service";
+import { LocalizationService } from "@terramatch-microservices/common/localization/localization.service";
+
+@Injectable()
+export class ResetPasswordService {
+  private readonly logger: Logger;
+
+  constructor(
+    private readonly jwtService: JwtService,
+    private readonly emailService: EmailService,
+    private readonly localizationService: LocalizationService
+  ) {
+    this.logger = new Logger(ResetPasswordService.name);
+  }
+
+  async sendResetPasswordEmail(emailAddress: string, callbackUrl: string) {
+    const user = await User.findOne({ where: { emailAddress } });
+    if (user == null) {
+      throw new NotFoundException('User not found');
+    }
+
+    const bodyKey = 'reset-password.body';
+    const subjectKey = 'reset-password.subject';
+
+    const localizationKeys = await this.localizationService.getLocalizationKeys([bodyKey, subjectKey]);
+
+    if (!localizationKeys.length) {
+      throw new NotFoundException('Localizations not found');
+    }
+
+    const bodyLocalization = localizationKeys.find(x => x.key == bodyKey);
+    const subjectLocalization = localizationKeys.find(x => x.key == subjectKey);
+
+    if (bodyLocalization == null) {
+      throw new NotFoundException('Localization body not found');
+    }
+
+    if (subjectLocalization == null) {
+      throw new NotFoundException('Localization subject not found');
+    }
+
+    const resetToken = await this.jwtService.signAsync({ sub: user.uuid }, { expiresIn: "2h" });
+
+    const bodyEmailContent = await this.localizationService.translate(bodyLocalization.value, user.locale);
+    const resetLink = `${callbackUrl}/${resetToken}`;
+    const bodyEmail = this.formatBody(bodyEmailContent, resetLink);
+    await this.emailService.sendEmail(user.emailAddress, subjectLocalization.value, bodyEmail);
+
+    return { email: user.emailAddress, uuid: user.uuid, userId: user.id };
+  }
+
+  private formatBody(bodyEmailContent: string, resetLink: string) {
+    const anchor = `<a href="${resetLink}" target="_blank">link</a>`;
+    return bodyEmailContent.replace('link', anchor).replace('enlace', anchor).replace('lien', anchor);
+  }
+
+  async resetPassword(resetToken: string, newPassword: string) {
+    let userGuid;
+    try {
+      const payload = await this.jwtService.verifyAsync(resetToken);
+      userGuid = payload.sub;
+    } catch (error) {
+      this.logger.error(error);
+      throw new BadRequestException('Provided token is invalid or expired');
+    }
+
+    const user = await User.findOne({ where: { uuid: userGuid } });
```

Comment:
I would use `attributes: ["id", "uuid"]` here since you're not using anything else from the model.